src/util/toBigNumberString.ts

```diff
@@ -8,7 +8,7 @@ export function toBigNumberString(n: number | string): string {
   const match = out.match(expoRegex)
 
   if (match != null) {
-    const exponent = parseInt(match[1])
+    const exponent = parseInt(match[0].replace('e', ''))
```

Comment:
You can keep the `match[1]` and change the matching group in the regex from `/e(?:\+|-)(\d+)$/` to `/e((?:\+|-)\d+)$/`.